orcid-web/src/main/webapp/static/javascript/angularOrcid.js

```diff
@@ -1669,8 +1669,12 @@ orcidNgModule.filter('workExternalIdentifierHtml', function(){
         else link = workIdLinkJs.getLink(id,type); 
         	
         if (link != null){
-        	if(link.lastIndexOf('http://') === -1 && link.lastIndexOf('https://') === -1) {        	    
-        		link = 'http://' + link;
+        	if(link.lastIndexOf('http://') === -1 && link.lastIndexOf('https://') === -1) {
+        	    if (link.lastIndexOf('//') > -1){
```

Comment:
Pretty close. `//` means the same protocol as the current url. `'https:'==document.location.protocol?'https:':'http:'`, should work nicely.